packages/gatsby/src/redux/plugin-runner.ts

```diff
@@ -40,12 +40,28 @@ interface ICreatePageAction {
 }
 
 export const startPluginRunner = (): void => {
-  emitter.on(`CREATE_PAGE`, (action: ICreatePageAction) => {
-    const page = action.payload
-    apiRunnerNode(
-      `onCreatePage`,
-      { page, traceId: action.traceId, parentSpan: action.parentSpan },
-      { pluginSource: action.plugin.name, activity: action.activity }
-    )
+  const plugins = store.getState().flattenedPlugins
+  const implementingPlugins = plugins.filter(plugin => {
+    const includesOnCreatePage = plugin.nodeAPIs.includes(`onCreatePage`)
+    if (includesOnCreatePage && process.env.GATSBY_EXPERIMENTAL_NO_PAGE_NODES) {
+      if (plugin.name === `internal-data-bridge`) {
+        return false
+      } else {
+        return includesOnCreatePage
+      }
+    }
+
+    return includesOnCreatePage
   })
+
+  if (implementingPlugins.length > 0) {
+    emitter.on(`CREATE_PAGE`, (action: ICreatePageAction) => {
+      const page = action.payload
+      apiRunnerNode(
+        `onCreatePage`,
+        { page, traceId: action.traceId, parentSpan: action.parentSpan },
+        { pluginSource: action.plugin.name, activity: action.activity }
+      )
+    })
+  }
```

Comment:
this doesn't guarantee that `onCreatePage` API won't run in `internal-data-bridge` plugin - it will run if any other plugin implement `onCreatePage`

Reply:
right — I'm not trying to disable it altogether — it still has its place. But if we can avoid function calls, let's avoid functional calls. Just invoking an API costs 3-4s / 100k invocations. Probably can make that a lot cheaper (which is what I'm looking at right now) but still #savealltheseconds

Reply:
ah nvm, yeah I need to block creating nodes in the internal-data-bridge as well 👍